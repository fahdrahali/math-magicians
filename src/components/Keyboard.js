import { checkPropTypes } from 'prop-types';
import React, { Component } from 'react';

class Keyboard extends Component {
  getValue = (e) => {
    const { getValue } = this.props;
    getValue(e);
  };

  getResult = (e) => {
    const { getResult } = this.props;
    getResult(e);
  };

  render() {
    const calculator = [
      'AC',
      '+/-',
      '%',
      'รท',
      '7',
      '8',
      '9',
      'x',
      '4',
      '5',
      '6',
      '-',
      '1',
      '2',
      '3',
      '+',
    ];
    return (
      <>
        {calculator.map((item) => (
          <button type="button" onClick={this.getValue} value={item} className="item" key={item}>
            {item}
          </button>
        ))}
        <button type="button" onClick={this.getValue} className="item num-0" value="0">
          0
        </button>
        <button type="button" onClick={this.getValue} className="item" value=".">
          .
        </button>
        <button type="button" onClick={this.getResult} className="item orange" value="=">
          =
        </button>
      </>
    );
  }
}

Keyboard.propTypes = { getValue: checkPropTypes.isRequired };
Keyboard.propTypes = { getResult: checkPropTypes.isRequired };

export default Keyboard;
