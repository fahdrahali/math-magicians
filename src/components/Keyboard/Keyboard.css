.item {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 60px;
    background-color: #f5f4f1;
    border: 1px solid #dfdddd;
    font-size: 22px;
    padding: 10px;
  }
  
  .num-0 {
    grid-column: 1 / 3;
  }
  
  .item:nth-child(4n-3) {
    background-color: #eca013;
  }
  
  .orange {
    background-color: #eca013;
  }
  