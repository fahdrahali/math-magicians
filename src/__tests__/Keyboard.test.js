import React from 'react';
import TestRenderer from 'react-test-renderer';
import Keyboard from '../components/Keyboard';

it('renders correctly', () => {
  const renderer = TestRenderer.create(<Keyboard />).toJSON();
  expect(renderer).toMatchSnapshot();
});
