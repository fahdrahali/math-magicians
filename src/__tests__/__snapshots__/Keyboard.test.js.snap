// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="AC"
  >
    AC
  </button>,
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="+/-"
  >
    +/-
  </button>,
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="%"
  >
    %
  </button>,
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="รท"
  >
    รท
  </button>,
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="7"
  >
    7
  </button>,
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="8"
  >
    8
  </button>,
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="9"
  >
    9
  </button>,
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="x"
  >
    x
  </button>,
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="4"
  >
    4
  </button>,
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="5"
  >
    5
  </button>,
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="6"
  >
    6
  </button>,
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="-"
  >
    -
  </button>,
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="1"
  >
    1
  </button>,
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="2"
  >
    2
  </button>,
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="3"
  >
    3
  </button>,
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="+"
  >
    +
  </button>,
  <button
    className="item num-0"
    onClick={[Function]}
    type="button"
    value="0"
  >
    0
  </button>,
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="."
  >
    .
  </button>,
  <button
    className="item orange"
    onClick={[Function]}
    type="button"
    value="="
  >
    =
  </button>,
]
`;
